pkg/inspector/inspector.go

```diff
@@ -0,0 +1,156 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package inspector
+
+import (
+	"context"
+	"sync"
+
+	"github.com/conduitio/conduit/pkg/foundation/log"
+	"github.com/conduitio/conduit/pkg/record"
+	"github.com/google/uuid"
+)
+
+// Session wraps a channel of records and provides:
+// 1. a way to send records to it asynchronously
+// 2. a way to know if it's closed or not
+type Session struct {
+	C chan record.Record
+
+	id          string
+	logger      log.CtxLogger
+	onClose     func()
+	closeReason error
```

Comment:
I wonder why we store this error, it's always going to be `context.Canceled` 🤔

Reply:
When I was working on this initially, I was pretty sure something could go wrong.:) I'll remove the redundant field, thanks for noticing!